### ISSUE TITLE
Android app extension research

### ISSUE THREAD

**Issue by qrluke** [qrlk/adwhore.net#5]
Need to find ways to implement ADWHORE.NET extension for YouTube app or some open source client

**Comment by Stimmenhotel** [comment]
Maybe the devs of Youtube vanced or smart Youtube can help you with that and maybe even included into their modded clients.

https://github.com/yuliskov/SmartYouTubeTV

https://vanced.app/

**Comment by qrlk** [comment]
Vanced recently [integrated SponsorBlock](https://github.com/YTVanced/SponsorBlock) so I don't think they'll be interested in my immature project. I'm leaning more towards maintaining a simple plugin over YouTube / Vanced that implements getVideoData calling and skipping segments, I don't have enough competence for submitting UI etc.

But before the release of the android client, there is still a lot of work to be done on the project.

**Comment by Stimmenhotel** [comment]
Haven't heard about sponsorblock yet and just tested it. I have to say I like your design (segment tools and addon icon popup) much more than SBs. For my submissions I really try to get a clean cut and that seems not possible with SB. :(

> I'm leaning more towards maintaining a simple plugin over YouTube / Vanced that implements getVideoData calling and skipping segments.

You mean a client that runs parralel and alters the playback of the offical yt client? That would be quite awesome. (I have very little knowlege in android dev. so I dont know that getVideoData and stuff is)

**Comment by qrlk** [comment]
getVideoData is just an ADN API call that returns all known segments per video.

I think it would be a separate APK with the functionality of skipping segments (but without submitting new ones) based on original client / Vanced. But I have to finish my desktop solution before extending ADN to other platforms.

**Comment by whoisroot** [comment]
I don't know if you are still interested in extending into android, but I think [NewPipe](https://github.com/TeamNewPipe/NewPipe) is the perfect candidate for receiving the ad information collected by ADN.

I don't know what would be better, forking it and extending the app while mantaining the ADN database private or working with NewPipe's developers to integrate ADN into it natively. Probably the android interface is not so friendly to submit segments, but maybe it's still worth trying.

**Comment by qrluke** [comment]
Yes, I'm still interested in android support, but that's a bit of a lower priority right now, as development of the main platform on which the rework of the extension will be based is still in progress. MVP's audience is in Russia, where the threat of blocking YouTube remains, which means that I will have to urgently add support for other platforms that the Kremlin will try to promote as an alternative. There will be support for android in any case, just after the problems that repel users from using the extension are resolved and it can finally be promoted to the masses.

I remember that the position of the developers of Newpipe regarding the blocking of sponsored integrations was that they do not accept the approach of not distinguishing between ethical and unethical advertising. But it was a long time ago. The ability to distinguish between ethical and unethical is the essence of my project. I wonder what their position is now and how, in principle, it will be possible to adapt to their moral compass. Starting with a project naming I guess. Probably have to start by renaming the project.

In any case, I will expand the list of platforms after the main problems that have emerged over the 2 years of operation of the MVP are resolved. I think these are the main things that will be ready after Christmas. Regarding the release: either it will be one huge update right after the conflict in Ukraine is resolved (which has already incredibly hurt the project), or if everything drags on, it will be something like early access with a gradual release of all components and a bunch of bugs.   
It is very difficult for me to decide how to develop the project, because what I am working on requires a certain level of freedom of speech, which is suppressed every day in Russia due to obvious reasons.

The project database will be transparent, there is no longer any reason to keep it private. I'm not sure that I can immediately provide the ability to download the database, but all project data will be transparent, except for those that need to be kept private to ensure the privacy of my users.
At the MVP stage, the idea was to partner with some content creators, giving them access to the database, so that they would make content ridiculing the hypocrisy of other bloggers in order to draw public attention to the problem. Since that time, it turned out that on the English-speaking Internet the problem of such hypocritical advertising is not as common as I thought, because Western states take on the role of regulating people to disclose that they are being sponsored. Now the Russians are trying to implement such things, so far they are doing some kind of monopolistic oligarchic garbage, but I think the effect of this will sooner or later be noticeable.

Regarding adding segments using the Android interface: I am against this approach. I need high accuracy and focus on interested contributors with a high reputation, not the crowd. Plus, I work on this project alone and finance it with my own pennies, so it will not be possible to support such things in the long term, since the complication will inevitably entail a drain on resources.
Most likely it will be a simple client that will simply anonymously download segments for the video being watched.

Thanks anyway, I'll keep this project in mind. Sorry for so much text.
